### PR TITLE
Add support for experimental IO features in hadd

### PR DESCRIPTION
**Pull request by bbockelm**
This PR adds a support for passing along the IO feature flags in the merger and allows access from the command-line for `hadd`.

Additionally, makes the `TIOFeatures` a serialized member of `TTree` and `TBranch`, allowing them to behave as expected when cloned.

### CODE REVIEW

File: main/src/hadd.cxx [hunk 1]
```diff
@@ -289,6 +289,7 @@ int main( int argc, char **argv )
                }
             }
          }
+         ++ffirst;
```
Comment (pcanal) [inline main/src/hadd.cxx]: In what circumstance does this bug manifest?

Comment (bbockelm) [reply main/src/hadd.cxx]: If you use the (unreleased) `-experimental-io-features` flag that accidentally snuck into #1003.

Comment (pcanal) [reply main/src/hadd.cxx]: The commit should really mention this :).  It sounds worth otherwise :)

**Comment by bbockelm** [comment]
All the unit test failures appear to be due to the extra bytes serialized in the tree.  Working on an equivalent `roottest` PR.

**Comment by pcanal** [comment]
See https://github.com/root-project/roottest/pull/101

File: tree/tree/src/TTree.cxx [hunk 2]
```diff
@@ -3070,6 +3070,9 @@ TTree* TTree::CloneTree(Long64_t nentries /* = -1 */, Option_t* option /* = "" *
    //       a chain we get the chain's current tree.
    TTree* thistree = GetTree();
 
+   // We will use this to override the IO features on the cloned branches.
+   ROOT::TIOFeatures features = this->GetIOFeatures();;
```
Comment (pcanal) [inline tree/tree/src/TTree.cxx]: Redudant semi colon.

**Comment by bbockelm** [comment]
@pcanal - selected subset of `clang-format` suggestions have been applied.

**Comment by bbockelm** [comment]
@pcanal - anything remaining on this one?  I don't think I can make Jenkins happy unless things merge simultaneously...

**Pull request merged**